### PR TITLE
feat: add SARIF support for snyk open-source

### PR DESCRIPTION
**Pull request by admons**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Add "--sarif" option to Snyk Open Source

Here's a screenshot from [Github code scanning with the result of the --sarif run](https://github.com/admon-synk/LOKI-121-sarif/security/code-scanning)

image

### CODE REVIEW

File: src/cli/commands/test/open-source-sarif-output.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  };
+}
+
+export function getRules(testResult: TestResult): sarif.ReportingDescriptor[] {
+  return _.chain(testResult.vulnerabilities)
+    .groupBy('id')
+    .values()
+    .map(
+      ([vuln, ...moreVulns]): sarif.ReportingDescriptor => {
+        const cve = vuln.identifiers?.CVE?.[0];
```
Comment (orsagie) [inline src/cli/commands/test/open-source-sarif-output.ts:56]: Do we expect all the CVE's to be the same?

Comment (admons) [reply src/cli/commands/test/open-source-sarif-output.ts:56]: No, thanks for that!
I just used what exists today on the IaC.
Now I changed it to show all CVEs

File: src/cli/commands/test/formatters/format-test-results.ts [hunk 2]
```diff
@@ -63,9 +64,13 @@ export function extractDataToSendFromResults(
   let sarifData = {};
   let stringifiedSarifData = '';
   if (options.sarif || options['sarif-file-output']) {
-    sarifData = !options.iac
-      ? createSarifOutputForContainers(results)
-      : createSarifOutputForIac(results);
+    if (options.iac) {
+      sarifData = createSarifOutputForIac(results);
+    } else if (!options.docker) {
```
Comment (orsagie) [inline src/cli/commands/test/formatters/format-test-results.ts:69]: Can you make this 
```suggestion
    } else if (options.docker) {
```
I feel like the default should be createSarifOutputForOpenSource, and not for Containers.

**Review by orsagie — APPROVED** [review]
LGreatTM

**Pull request merged**